appengine/flexible/SocialAuth/Services/KmsDataProtectionProvider.cs

```diff
@@ -103,34 +90,24 @@ IDataProtector IDataProtectionProvider.CreateProtector(string purpose)
                 return cached;
             }
             // Create the crypto key:
-            var keyRingName = string.Format(
-                "projects/{0}/locations/{1}/keyRings/{2}",
-                _options.Value.ProjectId, _options.Value.Location,
-                _options.Value.KeyRing);
-            string rotationPeriod = string.Format("{0}s",
-                    TimeSpan.FromDays(7).TotalSeconds);
             CryptoKey cryptoKeyToCreate = new CryptoKey()
             {
-                Purpose = "ENCRYPT_DECRYPT",
-                NextRotationTime = DateTime.UtcNow.AddDays(7),
-                RotationPeriod = rotationPeriod
+                Purpose = CryptoKey.Types.CryptoKeyPurpose.EncryptDecrypt,
+                NextRotationTime = Timestamp.FromDateTime(DateTime.UtcNow.AddDays(7)),
+                RotationPeriod = Duration.FromTimeSpan(TimeSpan.FromDays(7))

```

Comment:
Actually, it can't be a one-off operation run by developers, because the full set of key is not known at compile time.

The KMS stuff is confusing because in this app, it's used for two distinct purposes:
1. Encrypting appsecrets.json.
2. Providing a DataProtectionProvider used for Anti-Forgery tokens and by facebook middleware.